pkg/apis/v1alpha1/subnet_types.go

```diff
@@ -0,0 +1,78 @@
+/* Copyright © 2021 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type SubnetStatusCondition string
+
+const (
+	SubnetReady    SubnetStatusCondition = "Ready"
+	SubnetNotReady SubnetStatusCondition = "NotReady"
+)
+
+// SubnetSpec defines the desired state of Subnet
+type SubnetSpec struct {
+	// Size of subnet based upon estimated workload count.
+	// Defaults to 64.
+	// +kubebuilder:default:=64
+	IPV4SubnetSize int `json:"ipv4SubnetSize,omitempty"`
+	// Access mode of subnet, accessible only from within VPC or from outside VPC.
+	// Defaults to private.
+	// +kubebuilder:default:=private
+	// +kubebuilder:validation:Enum=private;public;isolated
+	AccessMode string `json:"accessMode,omitempty"`
+	// Subnet CIDR.
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=2
+	IPAddresses []string `json:"ipAddresses,omitempty"`
```

Comment:
Why CIDR is a slice? Should not it be a single string (assuming we use two fields for v4 and v6 respectively)? What is the definition in NSX-T API?

Reply:
NSX Subnet API define ip_addresses as array for v4 and v6 as below
   ip_addresses:
      type: array
      maxItems: 2
      items:
        type: string
        format: ip-cidr-block
      title: CIDR

Reply:
Ok. Then let us follow NSX-T API. BTW, what is the use of "title" in the NSX-T API spec?

Reply:
I think it is a summary of the field. We can see in nsx policy api yaml files , e.g.
http://build-squid.eng.vmware.com/build/mts/release/bora-20375668/publish/linux-centos64-64/openapi/nsx_policy_api.yaml